plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java

```diff
@@ -225,6 +231,51 @@ private DeltaLakeColumnHandle buildColumnHandle(EntryType entryType, CheckpointS
         return new DeltaLakeColumnHandle(entryType.getColumnName(), type, OptionalInt.empty(), entryType.getColumnName(), type, REGULAR, Optional.empty());
     }
 
+    /**
+     * Constructs a TupleDomain which filters on a specific required primitive sub-column of the EntryType being
+     * not null for effectively pushing down the predicate to the Parquet reader.
+     */
+    private TupleDomain<HiveColumnHandle> buildTupleDomainColumnHandle(EntryType entryType, HiveColumnHandle column)
+    {
+        String field;
+        Type type;
+        switch (entryType) {
+            case COMMIT:
+            case TRANSACTION:
+                field = "version";
```

Comment:
How do we know which fields exist in the column and which ones can be non-null ? If it's defined somewhere in the code or in the delta spec, it would help to have a code comment pointing to it

Reply:
It's part of the delta spec, and we also enforce it when constructing action entries. For most of the entry types this is listed as optional/required in the description tables in the delta spec, except for the Transaction and Protocol actions. I've opened https://github.com/delta-io/delta/pull/1890 to make it clearer.

Actually, it looks like we are too strict when reading transaction entries, and require that the `lastUpdated` field is set, even though it is optional in the spec: https://github.com/trinodb/trino/blob/e920b4b1e1f091ed9d2ceec82e50a83996599544/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java#L521. It should not cause any issues, as I believe we don't use Transaction entries anywhere, but it would be good to fix.